cmd/mt-whisper-importer-reader/main.go

```diff
@@ -248,123 +205,113 @@ func shortAggMethodString(aggMethod whisper.AggregationMethod) string {
-						file,
-						w.Header.Metadata.AggregationMethod,
-						aggMethodStr,
-					))
-				}
-			}
-
-			rowKey = api.AggMetricKey(
-				md.Id,
-				aggMethodStr,
-				archiveInfo.SecondsPerPoint,
-			)
-		} else {
-			rowKey = md.Id
-		}
-
-		// only read archive if archiveIdx is in readArchives
-		if _, ok := readArchives[archiveIdx]; !ok && len(readArchives) > 0 {
-			continue
-		}
-
-		encodedChunks := make([]chunk.IterGen, 0)
-
-		if len(chunkSpans)-1 < archiveIdx {
-			// if we have more archives than chunk spans are specified, we simply use the last one
-			chunkSpan = chunkSpans[len(chunkSpans)-1]
-		} else {
-			chunkSpan = chunkSpans[archiveIdx]
-		}
-
-		points, err := w.DumpArchive(archiveIdx)
+	points := make(map[int][]whisper.Point)
+	for i := range w.Header.Archives {
+		p, err := w.DumpArchive(i)
 		if err != nil {
-			return nil, errors.New(fmt.Sprintf("ERROR: Failed to read archive %d in %q, skipping: %q", archiveIdx, file, err))
+			return res, fmt.Errorf("Failed to dump archive %d from whisper file %s", i, file)
 		}
+		points[i] = p
+	}
 
-		var point whisper.Point
-		var t0, prevT0 uint32
-		var c *chunk.Chunk
-
-		for _, point = range sortPoints(points) {
-			// this shouldn't happen, but if it would we better catch it here because Metrictank wouldn't handle it well:
-			// https://github.com/raintank/metrictank/blob/f1868cccfb92fc82cd853914af958f6d187c5f74/mdata/aggmetric.go#L378
-			if point.Timestamp == 0 {
-				continue
+	conversion := newConversion(w.Header.Archives, points, method)
+	for retIdx, retention := range schema.Retentions {
+		convertedPoints := conversion.getPoints(retIdx, uint32(retention.SecondsPerPoint), uint32(retention.NumberOfPoints))
+		for m, p := range convertedPoints {
+			rowKey := getRowKey(retIdx, md.Id, m, retention.SecondsPerPoint)
+			encodedChunks := encodedChunksFromPoints(p, uint32(retention.SecondsPerPoint), retention.ChunkSpan)
+			archives = append(archives, archive.Archive{
+				SecondsPerPoint: uint32(retention.SecondsPerPoint),
+				Points:          uint32(retention.NumberOfPoints),
+				Chunks:          encodedChunks,
+				RowKey:          rowKey,
+			})
+			if int64(p[len(p)-1].Timestamp) > md.Time {
+				md.Time = int64(p[len(p)-1].Timestamp)
 			}
+		}
+	}
 
-			t0 = point.Timestamp - (point.Timestamp % chunkSpan)
-			if prevT0 == 0 {
-				c = chunk.New(t0)
-				prevT0 = t0
-			} else if prevT0 != t0 {
-				log(fmt.Sprintf("Mark chunk at t0 %d as finished", prevT0))
-				c.Finish()
-
-				encodedChunks = append(encodedChunks, *chunk.NewBareIterGen(c.Bytes(), c.T0, chunkSpan))
+	res.AggregationMethod = uint32(w.Header.Metadata.AggregationMethod)
+	res.MetricData = *md
+	res.Archives = archives
+	return res, nil
+}
 
-				log(fmt.Sprintf("Create new chunk at t0 %d with chunk span %d", t0, chunkSpan))
-				c = chunk.New(t0)
-				prevT0 = t0
-			}
+func getRowKey(retIdx int, id, meth string, resolution int) string {
```

Comment:
let's not introduce yet another term. use one of the existing 2 (SecondsPerPoint or aggSpan)